src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/CaseLinkService.java

```diff
@@ -0,0 +1,111 @@
+package uk.gov.hmcts.ccd.domain.service.casedeletion;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.caseaccess.CaseLinkEntity;
+import uk.gov.hmcts.ccd.data.caseaccess.CaseLinkRepository;
+import uk.gov.hmcts.ccd.data.casedetails.CaseDetailsRepository;
+import uk.gov.hmcts.ccd.data.casedetails.DefaultCaseDetailsRepository;
+import uk.gov.hmcts.ccd.domain.model.casedeletion.CaseLink;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+@Slf4j
+@Service
+public class CaseLinkService {
+
+    private final CaseDetailsRepository caseDetailsRepository;
+    private final CaseLinkRepository caseLinkRepository;
+    private final CaseLinkMapper caseLinkMapper;
+
+    @Inject
+    public CaseLinkService(CaseLinkRepository caseLinkRepository,
+                           CaseLinkMapper caseLinkMapper,
+                           @Qualifier(DefaultCaseDetailsRepository.QUALIFIER)
+                                   CaseDetailsRepository caseDetailsRepository) {
+        this.caseLinkRepository = caseLinkRepository;
+        this.caseLinkMapper = caseLinkMapper;
+        this.caseDetailsRepository = caseDetailsRepository;
+    }
+
+    public void updateCaseLinks(Long caseId,
+                                String caseTypeId,
```

Comment:
NB: CaseType should be 'case type id of the case to which the case links', see New CASE_LINK table in scope document.

Reply:
Added a TODO in for this since this will need to be discussed (which casetype should be coming through) as we are currently bringing in the main case casetypeid - should this be the casetypeid of the linked case [790b9eb](https://github.com/hmcts/ccd-data-store-api/pull/1719/commits/790b9ebd63a212b252fd036ceb8cc804f1679593)

Reply:
`case_type_id` population refactored in #1941